Sara/mkt 367 rbac user management (#567)
returntocorp/semgrep-docs: docs/semgrep-app/user-management.md

---
slug: user-management 
append_help_link: true
title: Managing users and roles 
description: "Learn about roles, user management, and how to implement role-based access control in Semgrep App."
---

import MoreHelp from "/src/components/MoreHelp"

# Managing users and roles

Accounts enable you to manage access to Semgrep resources, such as scans and findings, with varying levels of collaboration and visibility.

Semgrep App has two types of accounts:

<dl>
    <dt>Personal account</dt>
    <dd>Every person who signs into Semgrep App is first signed in to a <strong>personal account</strong>. In a personal account, your findings, projects, private rules, and scans are visible only to you.</dd>
    <dt>Organization account</dt>

>     <dd>To collaborate with others, create an <strong>organization account</strong>. An organization account in Semgrep App requires an existing organization from your source code management (SCM) tool. Creating this type of account integrates Semgrep App into your organization. A user can have many organizations, provided that they are a member of that organization in their SCM.</dd>

Author: @adamkvitek

I'm going with "A user can be part of many Semgrep App organization accounts."

For the change from "An" to "The" - what's the difference? 😅

>     <dd>To collaborate with others, create an <strong>organization account</strong>. An organization account in Semgrep App requires an existing organization from your source code management (SCM) tool. Creating this type of account integrates Semgrep App into your organization. A user can have many organizations, provided that they are a member of that organization in their SCM.</dd>

Reviewer: Just that you already mentioned @s-santillan "an organization account" in the first sentence, you can make it "The organization account" in the other, because you already talked about it. :D But feel free to ignore, what you said about "A user can be part of many Semgrep App organization accounts." is good!

>     <dd>To collaborate with others, create an <strong>organization account</strong>. An organization account in Semgrep App requires an existing organization from your source code management (SCM) tool. Creating this type of account integrates Semgrep App into your organization. A user can have many organizations, provided that they are a member of that organization in their SCM.</dd>

Reviewer: Oh, I was wrong, there are more accounts, fixed my suggestion above. @s-santillan

</dl>

> Users from the same organization are able to sign into Semgrep App with their SCM's credentials and are automatically added to the organization account. Semgrep App is also able to detect the organization's repositories for scanning.

Reviewer: ```suggestion
Users from the same organization can sign into Semgrep App with their SCM's credentials and are automatically added to the organization account. Semgrep App can also detect the organization's repositories available for scanning.
```

> In organization accounts, collaborators can collectively manage Semgrep App. By default, users are able to: 

Reviewer: ```suggestion
In organization accounts, collaborators can collectively manage Semgrep App. By default, users can:
```

* Add projects to scan
* View and triage findings

> * Determine what rules to run and actions

Reviewer: (Blocking - finish a sentence)

Just an idea (if you wanted to say the following):
```suggestion
* Determine what rules to run and set up actions that Semgrep App will perform. 
```

Author: (manually commited this change)

* Manage tokens and other settings

> * View an organization's private rules

Reviewer: (Semi blocking, worth considering.)
Write full stops at the end of all list items. Unless the items in the list are just single words use full stops. These list items you included are still sentences.
See example in https://developers.google.com/style/lists?hl=en:
<img width="834" alt="Screenshot 2022-06-09 at 12 31 24" src="https://user-images.githubusercontent.com/26669307/172827534-ac90af13-2e64-4fee-82da-25e2311abd07.png">

Semgrep App can restrict features based on user roles. See [Controlling access through roles](#controlling-access-through-roles).

> By creating an organization account, teams are able to collaborate on rule writing and the management of repositories. They are able to enforce organization-wide standards and secure their repositories at scale.

Reviewer: ```suggestion
By creating an organization account, teams can collaborate on rule writing and the management of repositories. Organization accounts can enforce organization-wide standards and secure their repositories at scale.
```

## Controlling access through roles

![Screenshot of role-based access control ](../img/rbac-overview.png)<br />

> 

Reviewer: I could not comment on things outside of your change, but in user-management.md, check these sentences below:

Original:
```md
## Setting up RBAC

Semgrep App divides users into two roles:

* `admin`
* `member`

:::info
Users in organizations without RBAC enabled are `admin` by default.
:::
:::info
Community-tier (Free) users are `admin` by default.
:::
```

Suggesting:
```md
## Setting up role-based access control (RBAC)

Semgrep App divides users into two roles:

* `admin`
* `member`

:::info
Users in organizations without RBAC enabled are assigned an `admin` role by default.
:::

:::info
Community-tier (Free) users are assigned an `admin` role by default.
:::
```

Author: (Added in latest commit)
